service/history/historyEngine.go

```diff
@@ -351,7 +351,7 @@ Update_History_Loop:
 		// Start a timer for the decision task.
 		timeOutTask := context.tBuilder.AddDecisionTimoutTask(scheduleID, di.DecisionTimeout)
 		timerTasks := []persistence.Task{timeOutTask}
-		defer e.timerProcessor.NotifyNewTimer(timeOutTask.GetTaskID())
+		defer e.timerProcessor.NotifyNewTimer(timerTasks)
```

Comment:
All the inflight timer tasks are not buffered today, so we use this when to do the query(when that timer is about to fail). So if the transaction fails it is ok, we would do a additional batch query at that time. Once this change is stabilized, i can definitely add the change to buffer and move the ack level with out reading from disk.